scripts/pi-hole/speedtest/speedtest.sh

```diff
@@ -1,326 +1,82 @@
-        download="download"
-        upload="upload"
-    fi
-else
-    mkdir -p $(dirname $log)
-
-    start=$(date +"%Y-%m-%d %H:%M:%S")
-
-    if [[ $last == true && -f "$log" ]]; then
-        # Reuse existing results (useful for debugging)
-        mdebug "Reusing existing results: $log"
-    else
-        # Query Speedtest
-        # Test for no-pre-allocate
-        pre_allocate=$(speedtest-cli --help | grep "no-pre-allocate" | wc -l)
-        if [[ "${pre_allocate}" -ge 1 ]]; then
-            opts="$opts --no-pre-allocate"
-        fi
-
-        # Test for custom server
-        if [[ "${SPEEDTEST_SERVER}" =~ ^[0-9]+$ ]]; then
-            if [[ "${SPEEDTEST_SERVER}" -ge 1 ]]; then
-                # echo "Using custom server $SPEEDTEST_SERVER"
-                opts="$opts --server $SPEEDTEST_SERVER"
-            fi
-        fi
-
-        cmd="speedtest-cli $opts"
-        # echo $cmd
-        # exit
-        mdebug "Querying Speedtest using '$cmd'"
-        $cmd >$log
-        status=$?
-        mdebug "Exit code: $status"
-    fi
+}
 
+nointernet(){
     stop=$(date +"%Y-%m-%d %H:%M:%S")
+    echo "No Internet"
+    sqlite3 /etc/pihole/speedtest.db "insert into speedtest values (NULL, '${start}', '${stop}', 'No Internet', '-', '-', 0, 0, 0, 0, '#');"
+    exit 1
+}
 
-    last=$(tail -c 2 /tmp/$user/speedtest-csv.log)
-
-    # temp fix for failed run
-    if [[ $last != "g" ]]; then
-        echo "Connection Failed"
-        if [ $retry = false ]; then
-            echo "Retry after 30 sec  "
-            sleep 30
-            retry_cmd="--retry"
-            self_path="/var/www/html/admin/scripts/pi-hole/speedtest/speedtest.sh"
-            $self_path $retry_cmd
-        else
-            echo "Retry failed aborting"
-        fi
-        exit
-    fi
-
-    # Was speedtest-cli successful?
-    if [[ $status -ne 0 ]]; then
-        mecho "ERROR: '$cmd' failed (exit code $status)."
-        mecho $(cat /tmp/$user/speedtest-csv.log >/tmp/csv-error.log)
-        exit $status
-    fi
-
-    # Raw results
-    if [[ $debug == true ]]; then
-        mdebug "Raw results"
-        bfr=$(cat $log | tr '\n' '~')
-        echo "[DEBUG] - ${bfr}" | sed 's/~/\n[DEBUG] - /g' | head -n -1 1>&2
-    fi
-
-    # Parse
-    mdebug "Parsing results"
-    from=$(str_extract "Testing from ")
-    from_ip=$(echo $from | sed 's/.*(//g' | sed 's/).*//g')
-    from=$(echo $from | sed 's/ (.*//g')
-    server=$(str_extract "Hosted by ")
-    server_ping=$(echo $server | sed 's/.*: //g')
-    server=$(echo $server | sed 's/: .*//g')
-    server_dist=$(echo $server | sed 's/.*\[//g' | sed 's/\].*//g')
-    server=$(echo $server | sed 's/ \[.*//g')
-
-    download=$(str_extract "Download: ")
-    upload=$(str_extract "Upload: ")
-    if [[ $share == true ]]; then
-        share_url=$(str_extract "Share results: ")
+tryagain(){
+    if grep -q official <<< "$(/usr/bin/speedtest --version)"; then
+        apt-get install -y speedtest- speedtest-cli
```

Comment:
This might not be possible in all cases (armv7 and aarch64 don't have packages )

Reply:
`speedtest-cli` should not be installed since its the python version